### ISSUE TITLE
False positive `unbalanced-dict-unpacking`

### ISSUE THREAD

**Issue by mthuurne** [pylint-dev/pylint#9986]
### Bug description

```python
class C:
    def __init__(self):
        self.data = {}

    def add(self, name, value):
        self.data[name] = value

    def dump(self):
        if len(self.data) == 1:
            (name, value), = self.data.items()
            print(f"just one: {name} -> {value}")
        else:
            print("empty or multiple items")
```

### Command used

```shell
pylint testcase.py
```

### Pylint output

```python
testcase.py:10:12: W0644: Possible unbalanced dict unpacking with self.data.items(): left side has 1 label, right side has 0 values (unbalanced-dict-unpacking)
```

### Expected behavior

I would expect no message to be issued.

The problem goes away if `data` is initialized with 1 item in it.

Ideally, pylint would take into account that the length of the dictionary has just been checked. I imagine that wouldn't be easy to implement though.

A possible alternative would be to not issue this message when a dictionary might have had its contents changed after its initialization.

### Pylint version

```shell
pylint 3.3.1
astroid 3.3.4
Python 3.11.10 (main, Sep 09 2024, 17:03:08) [GCC]
```

### OS / Environment

Linux (openSUSE Tumbleweed)
